### PR TITLE
Game versioning

### PR DESCRIPTION
**Pull request by allenaerostar**
Game versioning with version update task job scheduling in the background.

### CODE REVIEW

File: dietstory-routing/game_version_updater/updateGameVersion.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        print("Updating game version started.")
+
+        prev_game_version = GameVersions.objects.get(is_live=1)
+        prev_game_version.is_live = 0
+        prev_game_version.save()
+
+        current_game_version = GameVersions.objects.get(major_ver=major_ver, minor_ver=minor_ver)
+        current_game_version.is_live = 1
+        current_game_version.save()
+
```
Comment (BenjixD) [inline dietstory-routing/game_version_updater/updateGameVersion.py]: can we make this 1 transaction? look into transaction.atomic

File: dietstory-routing/game_version_updater/updateGameVersion.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from game_asset_manager.models import GameVersions
+from django.utils import timezone
+
+
+def add_all_jobs(job_scheduler):
+
```
Comment (BenjixD) [inline dietstory-routing/game_version_updater/updateGameVersion.py]: use a less generic name for this method

File: dietstory-routing/game_asset_manager/views.py [hunk 3]
```diff
@@ -94,33 +97,120 @@ def get(self, request, *args, **kwargs):
...
-                {'message': 'No version of the game exists.'},
-                status=status.HTTP_404_NOT_FOUND)
+			try:
+				major_ver = game_version_form.cleaned_data.get('major_ver')
+				minor_ver = game_version_form.cleaned_data.get('minor_ver')
+				live_by = game_version_form.cleaned_data.get('live_by')
+
+				game_version = GameVersions(major_ver=major_ver, minor_ver=minor_ver, live_by=live_by)
+				game_version.save()
+
```
Comment (BenjixD) [inline dietstory-routing/game_asset_manager/views.py]: save only after scheduler adds the job

File: dietstory-routing/game_asset_manager/views.py [hunk 4]
```diff
@@ -94,33 +97,120 @@ def get(self, request, *args, **kwargs):
...
-        else:
-            return JsonResponse(
-                {'message': 'No version of the game exists.'},
-                status=status.HTTP_404_NOT_FOUND)
+			try:
+				major_ver = game_version_form.cleaned_data.get('major_ver')
+				minor_ver = game_version_form.cleaned_data.get('minor_ver')
+				live_by = game_version_form.cleaned_data.get('live_by')
+
+				game_version = GameVersions(major_ver=major_ver, minor_ver=minor_ver, live_by=live_by)
```
Comment (BenjixD) [inline dietstory-routing/game_asset_manager/views.py]: check live_by to be later than now

File: dietstory-routing/game_asset_manager/views.py [hunk 5]
```diff
@@ -94,33 +97,120 @@ def get(self, request, *args, **kwargs):
...
+				game_version.save()
+
+				#add this job to the job scheduler
+				version_update_scheduler.add_job(update_game_version, 'date', run_date=game_version.live_by, args=[game_version.major_ver, game_version.minor_ver])
```
Comment (BenjixD) [inline dietstory-routing/game_asset_manager/views.py]: we should also be allowed to submit versions with no live_by values. Ensure to not create a job for this case.

File: dietstory-routing/game_version_updater/updateGameVersion.py (hunk 1 above)
Comment (allenaerostar) [reply dietstory-routing/game_version_updater/updateGameVersion.py]: Yes this is possible and will do.

File: dietstory-routing/game_asset_manager/views.py (hunk 3 above)
Comment (allenaerostar) [reply dietstory-routing/game_asset_manager/views.py]: Not sure I agree with this one

File: dietstory-routing/game_asset_manager/views.py (hunk 5 above)
Comment (allenaerostar) [reply dietstory-routing/game_asset_manager/views.py]: Ok this is new, will add.

File: dietstory-routing/game_version_updater/updateGameVersion.py (hunk 2 above)
Comment (allenaerostar) [reply dietstory-routing/game_version_updater/updateGameVersion.py]: Will we have more jobs for update GameVersions in the future?

File: dietstory-routing/game_asset_manager/views.py (hunk 4 above)
Comment (allenaerostar) [reply dietstory-routing/game_asset_manager/views.py]: Will add this in validation!

File: dietstory-routing/game_asset_manager/views.py (hunk 3 above)
Comment (allenaerostar) [reply dietstory-routing/game_asset_manager/views.py]: We may run into an issue where the schedule job is finished before it is saved?

File: dietstory-routing/game_version_updater/updateGameVersion.py (hunk 2 above)
Comment (BenjixD) [reply dietstory-routing/game_version_updater/updateGameVersion.py]: likely not but always a possibility

File: dietstory-routing/game_asset_manager/views.py (hunk 3 above)
Comment (BenjixD) [reply dietstory-routing/game_asset_manager/views.py]: well consider the issues in either case:
- game version saved but no schedule was created (saving before job creation)
- scheduler ran before game_version saved (saving after job creation)

In the first case, we have an implicit failure, and likely unnoticed until after version live failed to change.

In the second case, we get an immediate failure with no side effects.

The scheduler should gracefully fail if the version doesn't exist.

**Pull request merged**